lib/clearance/user.rb

```diff
@@ -75,13 +75,13 @@ def reset_remember_token!
     def update_password(new_password)
       self.password_changing = true
       self.password = new_password
+      valid?
 
-      if valid?
+      if errors[:password].blank?
         self.confirmation_token = nil
         generate_remember_token
+        save(validate: false)
```

Comment:
Right. The only way I know of to do this is with `update_columns`.

This would be a breaking change, but since it seems like the save/validate lifecycle of users/passwords is different, one thing you could do is split out a new table/model to represent authentication tokens and make them belong to a user. That way, you could expire/delete old tokens and create new ones without touching the user/model table and its maze of validations.